File: leetcode-python-go/2341. 数组能形成多少数对/main.go
package main

import (
	"fmt"
)

func numberOfPairs(nums []int) []int {
	cnt := map[int]bool{}
	res := 0
	for _, num := range nums {
		cnt[num] = !cnt[num]
		if !cnt[num] {
			res++
		}
	}
	return []int{res, len(nums) - 2*res}
}

func main() {
	nums := []int{1, 3, 2, 1, 3, 2, 2}
	results := numberOfPairs(nums)
	fmt.Println(results)
}

File: leetcode-python-go/2341. 数组能形成多少数对/main.py
from typing import List
from collections import defaultdict

nums = [1,3,2,1,3,2,2]

def numberOfPairs(nums: List[int]) -> List[int]:
    cnt = defaultdict(bool)
    res = 0
    for num in nums:
        cnt[num] = not cnt[num]
        if not cnt[num]:
            res += 1
    return [res, len(nums) - 2 * res]

print(numberOfPairs(nums))